src/ruby/puppetserver-lib/puppet/server/puppet_config.rb

```diff
@@ -24,14 +24,19 @@ def self.initialize_puppet_settings(puppet_config:, require_config:, push_settin
     # to override, and whose values are the desired values for the settings.
     # Values that are not strings are omitted, allowing for keys in the
     # HashMap from puppetserver to have true values.
-    Puppet.initialize_settings(
-        puppet_config.reduce([]) do |acc, entry|
-          acc << "--#{entry[0]}"
-          acc << entry[1] if entry[1].kind_of?(String)
-          acc
-        end,
-        require_config, push_settings_globally
-    )
+    cli_flags = puppet_config.reduce([]) do |acc, entry|
+      acc << "--#{entry[0]}"
+      acc << entry[1] if entry[1].kind_of?(String)
+      acc
+    end
+
+    # We check parameter length because Method#arity returns -1 for varargs
```

Comment:
Maybe good to include *why* this check is necessary at all? or is that overkill?

Reply:
I think it is, but I'm the one who did it this way to begin with. : D   Or at least, I'd put the rationale for a patch in the commit, and non-obvious peculiarities of the implementation in code comments.

Opinions @Magisus, @tvpartytonight ?

Reply:
I think that you answered why you use length rather than arity, but you don't answer why you use length at all? Maybe something like 

> Check the method arguments because of additional arguments added in Puppet 6.13.0.

Reply:
That's fair. Then maybe I think the commit message could use a little more detail. 😝